File: typescript/src/features/codex/multi_file_context.ts
import * as fs from 'fs';
import * as path from 'path';

export class MultiFileContext {
    public gatherContext(directory: string, maxFiles: number = 10): string {
        let context = '';
        let fileCount = 0;

        const files = fs.readdirSync(directory);
        for (const file of files) {
            if (fileCount >= maxFiles) break;

            const fullPath = path.join(directory, file);
            if (fs.statSync(fullPath).isFile() && !file.startsWith('.')) {
                context += `\n--- File: ${file} ---\n`;
                context += fs.readFileSync(fullPath, 'utf8').substring(0, 1000); // Only first 1000 chars
                fileCount++;
            }
        }
        return context;
    }
}

File: rust/src/codex/multi_file_context.rs
use std::fs;
use std::path::Path;

pub struct MultiFileContext;

impl MultiFileContext {
    pub fn new() -> Self {
        MultiFileContext
    }

    pub fn gather_context(&self, directory: &str, max_files: usize) -> String {
        let mut context = String::new();
        let mut file_count = 0;

        if let Ok(entries) = fs::read_dir(directory) {
            for entry in entries.flatten() {
                if file_count >= max_files {
                    break;
                }

                let path = entry.path();
                if path.is_file() {
                    if let Some(file_name) = path.file_name().and_then(|n| n.to_str()) {
                        if !file_name.starts_with('.') {
                            if let Ok(content) = fs::read_to_string(&path) {
                                context.push_str(&format!("\n--- File: {} ---\n", file_name));
                                let snippet: String = content.chars().take(1000).collect();
                                context.push_str(&snippet);
                                file_count += 1;
                            }
                        }
                    }
                }
            }
        }
        context
    }
}